fix: GiteaPullRequestGenerator webhook support (#13201)

n9 opened the pull request:
Simple fix for GiteaPullRequestGenerator.

Fixes: https://github.com/argoproj/argo-cd/issues/13201

----

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

crenshaw-dev commented:
Thanks @n9! Would you mind adding unit tests?

n9 commented:
Ok. (Disclaimer: I am not a go developer. I've just copy-pasted existing tests.)

crenshaw-dev commented on applicationset/webhook/webhook_test.go:
```diff
@@ -333,6 +353,28 @@ func fakeAppWithGithubPullRequestGenerator(name, namespace, owner, repo string)
 	}
 }
 
+func fakeAppWithGiteaPullRequestGenerator(name, namespace, owner, repo string, api string) *argoprojiov1alpha1.ApplicationSet {
+	return &argoprojiov1alpha1.ApplicationSet{
```
```suggestion
func fakeAppWithGiteaPullRequestGenerator(name, namespace, owner, repo string, api string) *v1alpha1.ApplicationSet {
	return &v1alpha1.ApplicationSet{
```

n9 commented:
Any chance to get this merged?

n9 commented:
@crenshaw-dev I have updated the PR to the current master (and updated the code to use the newer version of webhooks). Could you please review it (to reduce the conflicts caused by rebase).

ishitasequeira reviewed (changes requested):
Thanks @n9, Just a small request to add one liner about why we use Github secrets in Gitea. Rest PR looks good.

n9 commented on applicationset/webhook/webhook.go:
```diff
@@ -74,6 +76,10 @@ func NewWebhookHandler(namespace string, argocdSettingsMgr *argosettings.Setting
 	if err != nil {
 		return nil, fmt.Errorf("Failed to get argocd settings: %v", err)
 	}
+	giteaHandler, err := gitea.New(gitea.Options.Secret(argocdSettings.WebhookGitHubSecret)) // reusing github secret
```
Sure. Force-pushed.

ishitasequeira reviewed (approved):
Thanks @n9 ! The changes LGTM!

ishitasequeira reviewed (approved):
I see some changes post my last approval. But changes lgtm!!

todaywasawesome reviewed (changes requested):
Looks like we need to fix broken tests before moving forward. @n9 are you able to update to fix the failing tests?

n9 commented:
@todaywasawesome Do you know what changed in ArgoCD (or deps) that caused tests to fail?
